lib/wickedbase.pm

```diff
@@ -858,17 +858,22 @@ sub skip_check_logs_on_post_run {
     shift->{skip_check_logs_on_post_run} = 1;
 }
 
+sub check_logs_cursor {
+    my $cursor = script_output(q(journalctl -o export -n 1 | tr -dc '\n|[[:print:]]' |  grep __CURSOR));
+    ($cursor) = ($cursor =~ /^__CURSOR=(.*)$/m);
+    return $cursor;
+}
+
 sub check_logs {
     my $self = shift;
     my $code = shift;
-    my $cursor = '';
+    my $cursor = $self->{pre_run_check_logs_cursor} // '';
 
     if (ref($code) eq 'CODE') {
-        $cursor = script_output(q(journalctl -o export -n 1 | tr -dc '\n|[[:print:]]' |  grep __CURSOR));
-        ($cursor) = ($cursor =~ /^__CURSOR=(.*)$/m);
-        $cursor = "-c '$cursor'";
+        $cursor = $self->check_logs_cursor();
         $code->();
     }
+    $cursor = "-c '$cursor'" if (length($cursor) > 0);
```

Comment:
I would drop it from here and just add `$cmd .= "-c '" . $self->{pre_run_check_logs_cursor} . "'"` in place where it is actually used  https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/15250/files#diff-e5a344b1982337ed8b1fcce6c2ec13224675b3db7733fcce7acb66c82a8e8d71R895

Reply:
I briefly checked `run_test_shell_script` and I failed to find where set of `__CURSOR` happens there so I don't see relation between `run_test_shell_script` and setting of `__CURSOR`